### ISSUE TITLE
Proposal: Markdown formatter

### ISSUE THREAD

**Issue by dkuku** [elixir-lang/ex_doc#1785]
I recently watched Jeremy Howard's video about llm-s https://youtu.be/jkrNMKz9pWU?t=4656, the part about [h2ogpt](https://github.com/h2oai/h2ogpt) and parsing documents was very interesting.
I think having some kind of text output would be useful for faster parsing by llm.
Markdown output would make most sense because it has almost no overhead when creating from existing documentation.

**Comment by josevalim** [comment]
We will need more context. :D

**Comment by dkuku** [comment]
> We will need more context. :D

I pressed enter on the title. Expanding it :)

**Comment by josevalim** [comment]
@dkuku oh, I see. Actually, I think ExDoc is an overkill for this use case. Something like `Application.spec(:my_app, :modules)` and then `Code.fetch_docs` on the modules will be enough to build a pretty good reference. The hard part in ExDoc is actually autolinking, generating sidebars, etc, none of which are a concern for your use case.

However, it is worth noting that I did try to provide RAG on top of ExDoc and it turns out API references are not much helpful. I generate embeddings for all of Elixir docs and asking "what is a string?" or "what is a charlist" would bring function references, such as String.to_charlist or Integer.to_string, which would ultimately not help answer the question.

In any case, if you want to try out these ideas, both RAG and fine-tuning, the direction I gave above is more than enough to get something up and running. If we have concrete examples of how those can be used and in a way making it part of ExDoc will be essential for it to work, then I will be glad to revisit this. :)
